types/document.d.ts

```diff
@@ -228,17 +230,17 @@ declare module 'mongoose' {
     toObject<T = Require_id<DocType>>(options?: ToObjectOptions): Require_id<T>;
 
     /** Clears the modified state on the specified path. */
-    unmarkModified(path: string): void;
+    unmarkModified<T extends FlatPath<DocType>>(path: T): void;
 
     /** Sends an updateOne command with this document `_id` as the query selector. */
     updateOne(update?: UpdateQuery<this> | UpdateWithAggregationPipeline, options?: QueryOptions | null): Query<any, this>;
 
     /** Executes registered validation rules for this document. */
-    validate(pathsToValidate?: pathsToValidate, options?: AnyObject): Promise<void>;
+    validate<T extends FlatPath<DocType>>(pathsToValidate?: T | T[], options?: AnyObject): Promise<void>;
     validate(options: { pathsToSkip?: pathsToSkip }): Promise<void>;
 
     /** Executes registered validation rules (skipping asynchronous validators) for this document. */
     validateSync(options: { pathsToSkip?: pathsToSkip, [k: string]: any }): Error.ValidationError | null;
-    validateSync(pathsToValidate?: pathsToValidate, options?: AnyObject): Error.ValidationError | null;
+    validateSync<T extends FlatPath<DocType>>(pathsToValidate?: T | T[], options?: AnyObject): Error.ValidationError | null;
```

Comment:
I'm a bit wary of using `FlatPath<>` by default because of potential infinite recursion issues. Any self-referencing objects will lead to `FlatPath<>` blowing up, which is one of the reasons why the MongoDB Node driver [no longer uses their `Join` helper by default](https://github.com/mongodb/node-mongodb-native/blob/2fbb715d7d67039a15b563ea881808a3e29602a7/src/mongo_types.ts#L425).